tensorboard/webapp/core/views/hash_storage_component.ts

```diff
@@ -45,29 +46,35 @@ export class HashStorageComponent implements OnInit, OnChanges, OnDestroy {
   @Output()
   onValueChange = new EventEmitter<{prop: ChangedProp; value: string}>();
 
-  private onHashChangedImpl() {
-    const activePluginId = this.deepLinker.getPluginId();
-
-    if (activePluginId !== this.activePluginId) {
-      this.onValueChange.emit({
-        prop: ChangedProp.ACTIVE_PLUGIN,
-        value: activePluginId,
-      });
-    }
-  }
+  private readonly ngUnsubscribe = new Subject<void>();
+  private readonly onHashChange: Observable<Event> = fromEvent(
+    window,
+    'popstate',
+    {passive: true}
+  ).pipe(takeUntil(this.ngUnsubscribe));
 
   ngOnInit() {
-    // Cannot use the tf_storage hash listener because it binds to event before the
-    // zone.js patch. According to [1], zone.js patches various asynchronos calls and
-    // event listeners to detect "changes" and mark components as dirty for re-render.
-    // When using tf_storage hash listener, it causes bad renders in Angular due to
-    // missing dirtiness detection.
-    // [1]: https://blog.angular-university.io/how-does-angular-2-change-detection-really-work/
-    window.addEventListener('hashchange', this.onHashChange);
+    // Cannot use the tf_storage hash listener because it binds to event before
```

Comment:
It could be helpful to modify the comment here to briefly mention both alternative implementations that you have tried.

For example:

// Note: A couple alternative implementations to using 'popstate' event that turn out to be buggy:
//   1. tf_storage hash listener: It binds to events before zone.js patches event listeners for change detection ([1])
//   2. 'hashchange' event: We observed that window.history.back() and window.history.forward() do not trigger 'hashchange' events after some calls to replaceState
//
// [1]: https://blog.angular-university.io/how-does-angular-2-change-detection-really-work/

Reply:
Done.